clcache.py

```diff
@@ -148,29 +148,28 @@ def cacheDirectory(self):
         return self.dir
 
     def clean(self, stats, maximumSize):
-        with self.lock:
-            currentSize = stats.currentCacheSize()
-            if currentSize < maximumSize:
-                return
+        currentSize = stats.currentCacheSize()
```

Comment:
Why did you pull this outside the lock? currentCacheSize() is a read operation that should be protected.